packages/langium/src/grammar/langium-grammar-validator.ts

```diff
@@ -316,6 +327,15 @@ export class LangiumGrammarValidator {
         }
     }
 
+    checkTypes(grammar: ast.Grammar, accept: ValidationAcceptor): void {
+        for (const typeInconsistency of validateTypes(grammar)) {
+            accept('error', `The inferred type ${typeInconsistency.typeName} is not consistent with its declared version.`, {
```

Comment:
This message is not helpful enough. We need to print exactly what properties are inconsistent and why (different type, property not declared).